services/cannon/src/Cannon/Run.hs

```diff
@@ -85,20 +83,9 @@ run o = do
   let Brig bh bp = o ^. brig
       baseUrl = BaseUrl Http (unpack bh) (fromIntegral bp) ""
       clientEnv = ClientEnv manager baseUrl Nothing defaultMakeClientRequest
-      -- Loop the request until we get an answer. This is helpful during integration
-      -- tests where services are being brought up in parallel.
-      getInitialFedDomains = do
-        runClientM getFedRemotes clientEnv >>= \case
-          Right strat -> pure strat
-          Left err -> do
-            print $ "Could not retrieve the latest list of federation domains from Brig: " <> show err
-            threadDelay $ o ^. domainUpdateInterval
-            getInitialFedDomains
-  fedStrat <- getInitialFedDomains
-  tEnv <- newTVarIO fedStrat
-  let callback :: FederationDomainConfigs -> IO ()
-      callback = atomically . writeTVar tEnv
-  updateDomainsThread <- Async.async $ updateDomains clientEnv callback
+  fedStrat <- getAllowedDomainsInitial clientEnv
+  ioref <- newIORef fedStrat
```

Comment:
```
  fedStrat <- newIORef =<< getAllowedDomainsInitial clientEnv
```

(`ioref` is not a very helpful name.)

Reply:
also, who's using `ioref`?  adding it to `Env` and watching for updates is future work, correct?  would that also happen in `updateDomainsThread`, right after `getAllowedDomainsLoop`?